### PR TITLE
Update User guide, Developer guide; Improved on JavaDoc and some test cases

### PR DESCRIPTION
**Pull request by Derek-Hardy**

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -556,6 +555,61 @@ Each `Person` can also keep a list of groups he/she enrols in. An `UniqueGroupLi
...
+and implementation of these commands.
+
+==== `join` command usage
+The `join` command updates relationship between a person and a group existed in the MeetingBook.
+The person specified by name now becomes a member of the group specified by title.
+This command is executed with the following syntax: +
+
+Syntax: `join n/[Name] g/[Group]` +
+
+Example: `join n/Derek g/CS2101`: This command make the person 'Derek' become a member of group 'CS2101'
```
Comment (jeffreyooi) [inline docs/DeveloperGuide.adoc]: I think the explanation of the example should remove "This command..." in front. It should start with a verb, something like:
Makes person with the name "Derek" a member of group with title "CS2101"

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -102,7 +102,33 @@ Deletes the second person in the person list.
...
+=== Adding a new group: `addGroup`
+
+Add a new group with user input title into MeetingBook. +
+
+Format: `addGroup n/[Name]` +
+
+Examples:
+
+* `addGroup n/CS2103T` +
+Added a new group with title 'CS2103T' into the MeetingBook.
```
Comment (jeffreyooi) [inline docs/UserGuide.adoc]: Should be in present tense, e.g. Adds a new group with title 'CS2103T' into the MeetingBook.
Same for other examples.

File: docs/DeveloperGuide.adoc (hunk 1 above)
Comment (Derek-Hardy) [reply docs/DeveloperGuide.adoc]: OK Thanks for the suggestion!

File: docs/UserGuide.adoc (hunk 2 above)
Comment (Derek-Hardy) [reply docs/UserGuide.adoc]: Will correct them soon!

**Pull request merged**